vm.go

```diff
@@ -0,0 +1,210 @@
+	CompactUnevictableAllowed int64   // /proc/sys/vm/compact_unevictable_allowed
+	DirtyBackgroundBytes      int64   // /proc/sys/vm/dirty_background_bytes
+	DirtyBackgroundRatio      int64   // /proc/sys/vm/dirty_background_ratio
+	DirtyBytes                int64   // /proc/sys/vm/dirty_bytes
+	DirtyExpireCentisecs      int64   // /proc/sys/vm/dirty_expire_centisecs
+	DirtyRatio                int64   // /proc/sys/vm/dirty_ratio
+	DirtytimeExpireSeconds    int64   // /proc/sys/vm/dirtytime_expire_seconds
+	DirtyWritebackCentisecs   int64   // /proc/sys/vm/dirty_writeback_centisecs
+	DropCaches                int64   // /proc/sys/vm/drop_caches
+	ExtfragThreshold          int64   // /proc/sys/vm/extfrag_threshold
+	HugetlbShmGroup           int64   // /proc/sys/vm/hugetlb_shm_group
+	LaptopMode                int64   // /proc/sys/vm/laptop_mode
+	LegacyVaLayout            int64   // /proc/sys/vm/legacy_va_layout
+	LowmemReserveRatio        []int64 // /proc/sys/vm/lowmem_reserve_ratio
+	MaxMapCount               int64   // /proc/sys/vm/max_map_count
+	MemoryFailureEarlyKill    int64   // /proc/sys/vm/memory_failure_early_kill
+	MemoryFailureRecovery     int64   // /proc/sys/vm/memory_failure_recovery
+	MinFreeKbytes             int64   // /proc/sys/vm/min_free_kbytes
+	MinSlabRatio              int64   // /proc/sys/vm/min_slab_ratio
+	MinUnmappedRatio          int64   // /proc/sys/vm/min_unmapped_ratio
+	MmapMinAddr               int64   // /proc/sys/vm/mmap_min_addr
+	NrHugepages               int64   // /proc/sys/vm/nr_hugepages
+	NrHugepagesMempolicy      int64   // /proc/sys/vm/nr_hugepages_mempolicy
+	NrOvercommitHugepages     int64   // /proc/sys/vm/nr_overcommit_hugepages
+	NumaStat                  int64   // /proc/sys/vm/numa_stat
+	NumaZonelistOrder         string  // /proc/sys/vm/numa_zonelist_order
+	OomDumpTasks              int64   // /proc/sys/vm/oom_dump_tasks
+	OomKillAllocatingTask     int64   // /proc/sys/vm/oom_kill_allocating_task
+	OvercommitKbytes          int64   // /proc/sys/vm/overcommit_kbytes
+	OvercommitMemory          int64   // /proc/sys/vm/overcommit_memory
+	OvercommitRatio           int64   // /proc/sys/vm/overcommit_ratio
+	PageCluster               int64   // /proc/sys/vm/page-cluster
+	PanicOnOom                int64   // /proc/sys/vm/panic_on_oom
+	PercpuPagelistFraction    int64   // /proc/sys/vm/percpu_pagelist_fraction
+	StatInterval              int64   // /proc/sys/vm/stat_interval
+	Swappiness                int64   // /proc/sys/vm/swappiness
+	UserReserveKbytes         int64   // /proc/sys/vm/user_reserve_kbytes
+	VfsCachePressure          int64   // /proc/sys/vm/vfs_cache_pressure
+	WatermarkBoostFactor      int64   // /proc/sys/vm/watermark_boost_factor
+	WatermarkScaleFactor      int64   // /proc/sys/vm/watermark_scale_factor
+	ZoneReclaimMode           int64   // /proc/sys/vm/zone_reclaim_mode
+}
+
+// VM reads the VM statistics from the specified `proc` filesystem.
+func (fs FS) VM() (VM, error) {
```

Comment:
I think it would be better to use a `*VM` for the return type so that the struct doesn't need to be copied.

Reply:
Hm everything else returns a struct and not a pointer to struct. Why would VM be different?

Reply:
If you look at `class_power_supply.go` for example, `parsePowerSupply` returns a struct pointer, and `PowerSupplyClass()` returns a map which is also a reference type.  However, there seems to be a mix of both pointer and non-pointer return types for different functions.  In general, if a struct has more than a few fields then I'd tend to use a pointer, but maybe that's a premature optimization, so I'm ok with it if you prefer to leave as a struct.

Reply:
changed it to return `*VM`